### PR TITLE
Updated Getting Started Guide for vSphere Cloud Provider

### PR DESCRIPTION
**Pull request by divyenpatel**
This PR includes changes for the Getting Started Guide for vSphere Cloud Provider.

Following is the summary of change.

1. Added refined roles and permissions for vSphere Cloud Provider user.
2. Clarified steps for enabling vSphere Cloud Provider.
3. Removed deprecated content.

Please review : @BaluDontu @luomiao @tusharnt

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4418)

### CODE REVIEW

**Comment by divyenpatel** [comment]
rebased to sync with master branch. 
@jbeda @erictune Can you help with merging this PR

File: docs/getting-started-guides/vsphere.md [hunk 1]
```diff
@@ -40,201 +40,114 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
...
-
-#### Known issues
-
-### Kube-up (Deprecated)
 
-Kube-up.sh is no longer supported and is deprecated. The steps for kube-up are included but going forward [kube-anywhere](https://github.com/kubernetes/kubernetes-anywhere) is preferred.
+**[ Step 4 ]** Create and assign Roles to the vSphere Cloud Provider user and vSphere entities.
 
-The recommended version for kube-up is [v1.4.7](https://github.com/kubernetes/kubernetes/releases/tag/v1.4.7)
+vSphere Cloud Provider requires the following minimal set of privileges to interact with vCenter. Please refer [vSphere Documentation Center](http://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.vsphere.security.doc%2FGUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html&resultof=%22%43%72%65%61%74%65%22%20%22%63%72%65%61%74%22%20%22%43%75%73%74%6f%6d%22%20%22%63%75%73%74%6f%6d%22%20%22%52%6f%6c%65%22%20%22%72%6f%6c%65%22%20) to know about steps for creating a Custom Role, User and Role Assignment.
```
Comment (luomiao) [inline docs/getting-started-guides/vsphere.md]: This part is kind of complicated...
Is there a way to create a super user directly to include all the privileges, or it has to be done like the the following, to assign all those roles/privileges? Just curious...

File: docs/getting-started-guides/vsphere.md [hunk 2]
```diff
@@ -40,201 +40,114 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
...
-
-```shell
-govc import.vmdk kube.vmdk ./kube/
-```
-
-Verify that the VMDK was correctly uploaded and expanded to ~3GiB:
-
-```shell
-govc datastore.ls ./kube/
+Sample Config:
```
Comment (luomiao) [inline docs/getting-started-guides/vsphere.md]: For worker nodes, this config could be different.
If worker nodes set vm-name in the config file, they can exclude user/password in their config files.

File: docs/getting-started-guides/vsphere.md (hunk 1 above)
Comment (divyenpatel) [inline docs/getting-started-guides/vsphere.md]: Provided a Note:
`Note: if you want to use Administrator account then this step can be skipped.`

File: docs/getting-started-guides/vsphere.md (hunk 2 above)
Comment (divyenpatel) [inline docs/getting-started-guides/vsphere.md]: Provided two different conf file. and added note

`Note: vm-name parameter is introduced in 1.6.4 release. Both vm-uuid and vm-name are optional parameters. if vm-name is specified then vm-uuid is not used. if both are not specified then kubelet will get vm-uuid from /sys/class/dmi/id/product_serial and query vCenter to find the Node VM's name.`

**Comment by divyenpatel** [comment]
Addressed @luomiao review comments.

**Comment by divyenpatel** [comment]
I can not debug why `deploy/netlify` has failed.  I do not have access to `/sites/kubernetes-io-master-staging/deploys/59729197a700c44f9d5d1230 `

@jbeda @erictune can you help?

File: docs/getting-started-guides/vsphere.md [hunk 3]
```diff
@@ -40,201 +40,137 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
...
-export GOVC_GUEST_LOGIN='kube:kube'
-export GOVC_PORT='443'
-export GOVC_DATACENTER='Datacenter'
+**`vsphere.conf` for Master Node:**
+ ```
+[Global]
+        user = "vCenter username for cloud provider"
+        password = "password"
+        server = "IP/FQDN for vCenter"
+        port = "443" #Default 443 for vCenter
```
Comment (bradbeam) [inline docs/getting-started-guides/vsphere.md]: Isn't the `port` parameter being deprecated?

Comment (divyenpatel) [inline docs/getting-started-guides/vsphere.md]: Yes it is deprecated and not required in the vsphere.conf file. Removed it.

File: docs/getting-started-guides/vsphere.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ This page covers how to get started with deploying Kubernetes on vSphere and det
 * TOC
 {:toc}
 
-### Getting started with vSphere
+### Getting started with the vSphere
```
Comment (bradbeam) [inline docs/getting-started-guides/vsphere.md]: This seems incomplete, maybe `Getting started with the vSphere Cloud Provider`?

Comment (divyenpatel) [inline docs/getting-started-guides/vsphere.md]: fixed.

**Comment by luomiao** [comment]
LGTM

File: docs/getting-started-guides/vsphere.md [hunk 5]
```diff
@@ -40,201 +40,135 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
...
-```shell
-# Sample commands for v0.8.0 for 64 bit Linux.
-curl -OL https://github.com/vmware/govmomi/releases/download/v0.8.0/govc_linux_amd64.gz
-gzip -d govc_linux_amd64.gz
-chmod +x govc_linux_amd64
-mv govc_linux_amd64 /usr/local/bin/govc
-```
-
-* Get or build a [binary release](/docs/getting-started-guides/binary_release)
+vSphere Cloud Provider requires the following minimal set of privileges to interact with vCenter. Please refer [vSphere Documentation Center](http://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.vsphere.security.doc%2FGUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html&resultof=%22%43%72%65%61%74%65%22%20%22%63%72%65%61%74%22%20%22%43%75%73%74%6f%6d%22%20%22%63%75%73%74%6f%6d%22%20%22%52%6f%6c%65%22%20%22%72%6f%6c%65%22%20) to know about steps for creating a Custom Role, User and Role Assignment.
```
Comment (jbeda) [inline docs/getting-started-guides/vsphere.md]: Nit: this URL is ugly.  It seems to redirect to a much prettier: https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.vsphere.security.doc/GUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html

File: docs/getting-started-guides/vsphere.md [hunk 6]
```diff
@@ -40,201 +40,135 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
...
-export GOVC_INSECURE=1
-export GOVC_DATASTORE='datastore1'
-export GOVC_RESOURCE_POOL='/Datacenter/host/10.20.104.24/Resources'
-export GOVC_GUEST_LOGIN='kube:kube'
-export GOVC_PORT='443'
-export GOVC_DATACENTER='Datacenter'
+**`vsphere.conf` for Master Node:**
+ ```
+[Global]
+        user = "vCenter username for cloud provider"
```
Comment (jbeda) [inline docs/getting-started-guides/vsphere.md]: I assume that these are values that users can fill in.  Perhaps some sort of indication there like curly braces or something?

**Comment by jbeda** [comment]
/lgtm

Not sure what is going on with netlify.  Some nits on style but nothing critical.

File: docs/getting-started-guides/vsphere.md (hunk 5 above)
Comment (divyenpatel) [inline docs/getting-started-guides/vsphere.md]: Fixed.

**Comment by chenopis** [comment]
@divyenpatel It's definitely the `<br>` tag in the table that is causing the error during the preview build. I'm still researching a workaround.

**Comment by divyenpatel** [comment]
@chenopis Can we use `\s\s` in the Markdown instead of `<br>` tag as mentioned here - http://markdown-guide.readthedocs.io/en/latest/basics.html?highlight=break#line-return

**Comment by chenopis** [comment]
@divyenpatel It didn't seem to work for me when I tried it. I ended up just converting the table into HTML and using the <br> tag. You can preview the changes here: https://deploy-preview-4418--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/vsphere/

**Comment by chenopis** [comment]
@divyenpatel I fixed some other formatting issues, so if the preview looks good to you, I can go ahead and merge this.

**Comment by divyenpatel** [comment]
Thanks @chenopis for fixing md formatting issues. Preview looks good to me. I have squashed all commits and added known issues section. You can merge this PR.

**Pull request merged**